.github/workflows/build-full.yml

```diff
@@ -1,28 +1,17 @@
-name: ci
-
+name: full-build
 on:
-  push:
-    branches:
-      - develop
-      - backports/**
-  pull_request:
-    branches:
-      - develop
-      - backports/**
-    paths-ignore:
-      - '**.rst'
-      - '**.md'
-      - 'scripts/dev'
-  merge_group:
-    types: [checks_requested]
+  workflow_dispatch:
+  workflow_call:
 
 concurrency:
   group: ${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
   cancel-in-progress: true
 
 jobs:
-  build:
-    name: CI
+  # TODO: this currently includes non-GPU builds as well
+  # and won't actually run on device
+  gpu:
```

Comment:
Why name it gpu if it will also run on cpu?

Reply:
Once I have time I plan to only use these images for GPU builds, and have separate ones for cpu only. I think. Either way, they're both using images designed for nvidia-docker and rocm-docker.

Reply:
> Once I have time I plan to only use these images for GPU builds

Would it make sense to change the name when that happens?

Reply:
Right now they are GPU builds. We're just not running them on GPU.